run_dim_reduction.py

```diff
@@ -278,74 +299,98 @@ def parse_args():
-             "If left out the script will run with inference model and load saved model in 'weights'"
-             "to transform the vectors. Inference mode is only supported for PCA at the moment"
-    )
-    parser.set_defaults(fit_model=False)
-    parser.add_argument(
-        '-p', '--prefix',
-        type=str,
-        required=False,
-        default=None,
-        help='prefix of the latent vector filename "{}_latent_space"',
-    )
+    # parser.add_argument(
+    #     '-w', '--weights',
+    #     type=str,
+    #     required=True,
+    #     help="Directory to load/save the PCA weights.",
+    # )
+    # parser.add_argument(
+    #     '-f', '--fit',
+    #     dest='fit_model',
+    #     action='store_true',
+    #     help="Fit pca or umap model to the data and save the weights to 'weights'. "
+    #          "If left out the script will run with inference model and load saved model in 'weights'"
+    #          "to transform the vectors. Inference mode is only supported for PCA at the moment"
+    # )
+    # parser.set_defaults(fit_model=False)
+    # parser.add_argument(
+    #     '-p', '--prefix',
+    #     type=str,
+    #     required=False,
+    #     default=None,
+    #     help='prefix of the latent vector filename "{}_latent_space"',
+    # )
 
+    # parser.add_argument(
+    #     '-c', '--condition',
+    #     nargs='+',
+    #     type=str,
+    #     required=False,
+    #     default=None,
+    #     help='Condition for each input directory',
+    # )
     parser.add_argument(
-        '-c', '--condition',
-        nargs='+',
+        '-c', '--config',
         type=str,
-        required=False,
-        default=None,
-        help='Condition for each input directory',
+        required=True,
+        help='path to yaml configuration file'
     )
     return parser.parse_args()
 
 
 if __name__ == '__main__':
-    args = parse_args()
-
-    dim_reduction(input_dirs=args.input,
-                  output_dirs=args.output,
-                  weights_dir=args.weights,
-                  method=args.method,
-                  fit_model=args.fit_model,
-                  prefix=args.prefix,
-                  conditions=args.condition,
-                  )
+    arguments = parse_args()
+    config = YamlReader()
+    config.read_config(arguments.config)
+
+    # dim_reduction(input_dirs=args.input,
+    #               output_dirs=args.output,
+    #               weights_dir=args.weights,
+    #               method=args.method,
+    #               fit_model=args.fit_model,
+    #               prefix=args.prefix,
+    #               conditions=args.condition,
+    #               )
+
+    # batch run
+    for (input, output, weights, prefix, condition) in list(zip(config.dim_reduction.input_dirs,
+                                                                config.dim_reduction.output_dirs,
+                                                                config.dim_reduction.weights_dirs,
```

Comment:
`weights_dirs` shouldn't be zipped with `input_dirs` and `output_dirs` because normally there would only be one `weights_dir`. The script is designed to pool latent vectors from different directories and train one pca/umap model  on them. Let me know if microglia data is analyzed differently and we can make the script work for different analyses.

Reply:
>weights_dirs shouldn't be zipped with input_dirs and output_dirs because normally there would only be one weights_dir
Good point.  I think this only worked in my tests because I had exactly one of each file.

>The script is designed to pool latent vectors from different directories and train one pca/umap model on them. Let me know if microglia data is analyzed differently and we can make the script work for different analyses.

This is still done but on a "per well" basis.  The `file_name_prefixes` represents the well that is aggregated across all `input_dirs`.  However, I think my example passed because I used exactly one prefix (and exactly one input/output/weights).
But is this the behavior you want?  Do you want the `config.dim_reduction.input_dirs` to be a list to batch across (each element processed separately), or to be a list to aggregate across (extract and aggregate`file_name_prefix` from each element and process that)?

Reply:
> or to be a list to aggregate across (extract file_name_prefix from each element, aggregate them, then process)
The script was designed to have this behavior: concatenate all the latent vectors -> train a single model -> save the transformed vectors in each input dir (for PCA), or save the concatenated transform vectors in weight dir (UMAP). I find saving the concatenated transform vectors makes downstream analysis easier, but keep the behavior for PCA for backward compatibility.

Is this the behavior you want too?